scripts/compile-references.js

```diff
@@ -126,7 +126,10 @@ async function configureTypeScriptReferences(targetPackage, expectedReferences)
     /** @type {string[]} */
     const currentReferences = (tsconfigJson['references'] || []).map(reference => reference.path);
     // Compare both arrays: if an element is not the same we need to rewrite.
-    needRewrite = needRewrite || currentReferences.some((reference, index) => expectedReferences[index] !== reference);
+    needRewrite = needRewrite
+        || currentReferences.some((reference, index) => expectedReferences[index] !== reference)
+        || expectedReferences.some((reference, index) => currentReferences[index] !== reference);
+
```

Comment:
Nice catch, the previous logic didn't properly compare if the two arrays were equal.

Though I would suggest using the following logic instead:

```suggestion
    needRewrite = needRewrite
        || currentReferences.length !== expectedReferences.length
        || currentReferences.some((reference, index) => expectedReferences[index] !== reference);
```